File: app/src/main/java/com/test/murphy/weatherapp/WeatherApp.java
package com.test.murphy.weatherapp;

import android.app.Application;
import android.content.Context;

/**
 * Created by wsmurphy on 11/13/17.
 */

public class WeatherApp extends Application {

    private static Context mContext;
    private static WeatherApp mInstance;

    public WeatherApp() {
    }

    public static WeatherApp getInstance() {
        return mInstance;
    }

    @Override
    public void onCreate() {
        super.onCreate();

        mContext = getApplicationContext();
    }

    public static Context getContext() { return mContext; }
}

File: app/src/main/java/com/test/murphy/weatherapp/WeatherApp.kt
package com.test.murphy.weatherapp

import android.app.Application
import android.content.Context

class WeatherApp : Application() {

    override fun onCreate() {
        super.onCreate()

        context = applicationContext
    }

    companion object {

        @JvmStatic
        lateinit var context: Context
            private set

        @JvmStatic
        val instance by lazy { WeatherApp() }
    }


}
